specification/storage/resource-manager/Microsoft.Storage/stable/2021-06-01/storage.json

```diff
@@ -134,6 +134,9 @@
           },
           "StorageAccountCreateUserAssignedEncryptionIdentityWithCMK": {
             "$ref": "./examples/StorageAccountCreateUserAssignedEncryptionIdentityWithCMK.json"
+          },
+          "StorageAccountCreateWithImmutabilityPolicy":{
+            "$ref": "./examples/StorageAccountCreateWithImmutabilityPolicy.json"
```

Comment:
per my test, create account with account level WORM will fail with error `{"error":{"code":"InvalidStorageAccountImmutabilityPolicy","message":"Storage account level default immutability policy cannot be set as account level WORM is disabled"}}`, request ID `x-ms-request-id: b5293f9d-3fab-461a-9c9e-fa2b8b5901c8`

So if we should only set true to the "enabled" value (since update with "enabled" as false also fail), we might should not set this property in create/update account.

Reply:
This is by compliance. Customer has to explicitly enable Account level VLW - ImmutableStorageWithVersioning.enabled = true
Only then ImmutableStorageWithVersioning.immutabilityPolicy can be set. 
This is expected server behavior

ImmutableStorageWithVersioning.enabled is immutable and is mentioned in the description of the model

Reply:
This is not a good design:
If it can only be set to true, we might should not use bool as the type, to avoid unnecessary invalid user input. 
Or might should not allow user to input it, as it's single value. When user input an immutability policy, the value should set to true automatically. 
Besides that, the limitation for only true is allowed value, should be mentions in description.

Reply:
There is a gap here, immutabilityPolicy is an OPTIONAL FIeld. So we cannot get rid of enabled property

It is absolutely legal to set enabled=false to account creation time. 
By setting enabled=true , all the containers created under such an account will be VLW enabled and user can set ImmutabilityPolicy at the blob level, or optionally at container level using CLW apis or account level using this model.

Reply:
>  immutabilityPolicy is an OPTIONAL FIeld. So we cannot get rid of enabled property
[Wei] Actually, you don't need enable property:  In create account, just don't add the worm policy in create account request, will mean the account don't have account level worm policy. In update account, actually, the value should always be "true" when set worm policy, so add the enable property don't add any mire info to the update account rest request.

> It is absolutely legal to set enabled=false to account creation time.
[Wei] Server doesn't allow it, Set enabled=false to account creation , will get error `{"error":{"code":"InvalidStorageAccountImmutabilityPolicy","message":"Storage account level default immutability policy cannot be set as account level WORM is disabled"}}`

Reply:
Customers can simply enable account level WORM in CREATE and want to add Immutability Policy in UPDATE. We cannot have two different models for CREATE and UPDATE flows. 
enabled being immutable and immuatbilityPolicy being mutable

Reply:
Actually, per my test, in both create/update storage account, "enabled" must be true or server will report error.
If this is expect, "Enabled" is not a necessary parameter, as it must be true
If this is not expected, please check why create/update account  with "enabled" as false fail on server